Sources/CartonCLI/Commands/TestRunners/BrowserTestRunner.swift

```diff
@@ -12,39 +12,165 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import AsyncHTTPClient
 import CartonHelpers
 import CartonKit
 import Foundation
+import NIOCore
+import NIOPosix
 import PackageModel
 import TSCBasic
+import WebDriverClient
 
 private enum Constants {
   static let entrypoint = Entrypoint(fileName: "test.js", sha256: testEntrypointSHA256)
 }
 
+enum BrowserTestRunnerError: Error, CustomStringConvertible {
+  case invalidRemoteURL(String)
+  case failedToFindDriver
+
+  var description: String {
+    switch self {
+    case let .invalidRemoteURL(url): return "Invalid remote URL: \(url)"
+    case .failedToFindDriver:
+      return """
+      Failed to find WebDriver executable or remote URL to a running driver process.
+      Please make sure that you satisfied one of the followings (smaller item number lower priority):
```

Comment:
nit: I'd rephrase it this way:
```suggestion
      Please make sure that you are satisfied with one of the followings (in order fo priority):
```